PatternPal/PatternPal.StepByStep/InstructionSets/SingletonInstructionSet.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using PatternPal.StepByStep.Abstractions;
+using PatternPal.Core.Recognizers;
+using PatternPal.Recognizers.Abstractions;
+using PatternPal.Core.Checks;
+using PatternPal.Core.StepByStep;
+
+namespace PatternPal.StepByStep.InstructionSets
+{
+    public class SingletonInstructionSet : IInstructionSet
```

Comment:
Don't hard-code instruction sets for each pattern.

Instead, create an `InstructionSet` class which takes the recognizer as an argument. This means you also don't need the `InstructionSetsCreator`.